assets/scripts/palette/__tests__/UndoRedo.test.js

```diff
@@ -1,98 +1,103 @@
-    // It doesn't really matter what we set props to.
-    wrapper.setProps({
-      undoPosition: 1,
-      undoStack: []
+  it('enables undo button when an undo stack is created', () => {
+    const wrapper = renderWithReduxAndIntl(<UndoRedo />, {
+      initialState: {
+        undo: {
+          stack: [],
+          position: 0
+        }
+      }
     })
+    const undoButton = wrapper.getByTitle('Undo')
+    const redoButton = wrapper.getByTitle('Redo')
 
-    expect(isUndoAvailable).toBeCalled()
-    expect(isRedoAvailable).toBeCalled()
+    // Initial button states are disabled
+    expect(undoButton.disabled).toEqual(true)
+    expect(redoButton.disabled).toEqual(true)
 
-    // Clear mocks for future tests
-    isUndoAvailable.mockClear()
-    isRedoAvailable.mockClear()
-  })
+    // Update undo stack state from "elsewhere", and `<UndoRedo>` should
+    // receive new props. This is akin to a user editing the street once
+    wrapper.store.dispatch(replaceUndoStack([{ foo: 'bar' }], 1))
 
-  it('checks if undo or redo is available when undo stack changes', () => {
-    // `<UndoRedo>` is mounted with initial props
-    const wrapper = mount(<UndoRedo undoPosition={0} undoStack={[]} />)
+    // Undo button state should become enabled
+    // Redo button should remain disabled
+    expect(undoButton.disabled).toEqual(false)
+    expect(redoButton.disabled).toEqual(true)
+  })
 
-    // Send new props to `<UndoRedo>`
-    // It doesn't really matter what we set props to.
-    wrapper.setProps({
-      undoPosition: 0,
-      undoStack: [{ foo: 'bar' }]
+  it('enables redo button when undo stack position is lower than the top of the stack', () => {
+    const wrapper = renderWithReduxAndIntl(<UndoRedo />, {
+      initialState: {
+        undo: {
+          stack: [
+            { foo: 'bar' },
+            { foo: 'baz' }
+          ],
+          position: 0
+        }
+      }
     })
 
-    expect(isUndoAvailable).toBeCalled()
-    expect(isRedoAvailable).toBeCalled()
-
-    // Clear mocks for future tests
-    isUndoAvailable.mockClear()
-    isRedoAvailable.mockClear()
+    // In this scenario, redo is available, but undo is not.
+    expect(wrapper.getByTitle('Undo').disabled).toEqual(true)
+    expect(wrapper.getByTitle('Redo').disabled).toEqual(false)
   })
 
-  it('renders undo button enabled', () => {
-    // `<UndoRedo>` is mounted with initial props
-    const wrapper = mount(<UndoRedo undoPosition={0} undoStack={[]} />)
-
-    // Initial button state at mount time is disabled
-    const disabled1 = wrapper.find('button').last().prop('disabled')
-    expect(disabled1).toEqual(true)
-
-    // Send new props to `<UndoRedo>`
-    // It doesn't really matter what we set props to. The mock implementation
-    // of `isUndoAvailable` currently only returns `true`
-    wrapper.setProps({
-      undoPosition: 1,
-      undoStack: [{ foo: 'bar' }]
+  it('calls undo action when undo button is clicked', () => {
```

Comment:
I would change the description to your comment and then remove the comment:

> Expect the undo position to decrement when undo button is clicked

Reply:
or maybe even test something different.
e.g. now that the undo button is clicked, the redo button should be available, right?